### PR TITLE
wang_corpus_v1

### PR DESCRIPTION
**Pull request by CoderAlison**

### CODE REVIEW

File: exercise-001-corpus/corpus/corpus.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+type Pair struct {
+	Word string
+	Freq int
+}
+
+type PairList []Pair
+
+// implement sort interface
+func (this PairList) Len() int {
```
Comment (diwant) [inline exercise-001-corpus/corpus/corpus.go]: I see you have `this` as the name of the receiver variable.  Go Best Practices suggest not naming receiver variables as `this` or `self`.

See https://github.com/golang/go/wiki/CodeReviewComments#receiver-names

File: exercise-001-corpus/corpus/corpus.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+func (this PairList) Swap(i, j int) {
+	this[i], this[j] = this[j], this[i]
+}
+
+func sortByFreq(dict map[string]int) PairList {
+	sorted := make(PairList, len(dict))
+	i := 0
+	for word, freq := range dict {
```
Comment (diwant) [inline exercise-001-corpus/corpus/corpus.go]: From `dict` I see you are from the python world! *fist bump*

File: exercise-001-corpus/corpus/corpus.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		sorted[i] = Pair{word, freq}
+		i++
+	}
+	sort.Sort(sort.Reverse(sorted))
```
Comment (diwant) [inline exercise-001-corpus/corpus/corpus.go]: Why do you reverse and then sort?

File: exercise-001-corpus/corpus/corpus.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	return sorted
+}
+
+func WordCount(s string) PairList {
+	words := strings.Fields(s)
```
Comment (diwant) [inline exercise-001-corpus/corpus/corpus.go]: Punctuation would be included in words here.  So, 

> If this were your test statement, then your tests would say the word 'were' appears once whereas it appears thrice, as it were.

File: exercise-001-corpus/corpus/corpus_test.go [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestCorpus(t *testing.T) {
+	result := WordCount("Are you serious? I knew you were.")
+	assert.Equal(t, len(result), 6)
+	assert.Equal(t, result[0].Word, "you")
+	assert.Equal(t, result[0].Freq, 2)
```
Comment (diwant) [inline exercise-001-corpus/corpus/corpus_test.go]: This demonstrates your understanding of setting up tests in Go, which is great.  I would Include tests for weird/edge cases as well, such as words that should have zero count, punctuation-adjacent words, and capital and lowercase combos (which you have taken care of in your code so the test will pass)

File: exercise-001-corpus/corpus/corpus.go [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+)
+
+type Pair struct {
+	Word string
+	Freq int
+}
+
+type PairList []Pair
+
+// implement sort interface
```
Comment (diwant) [inline exercise-001-corpus/corpus/corpus.go]: Comments on exported functions should be,
```
// FunctionName ...
```

File: exercise-001-corpus/corpus/corpus.go [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+// deterministic, both Freq and Word should be taken into consideration
+func (this PairList) Less(i, j int) bool {
+	if this[i].Freq == this[j].Freq {
+		return this[i].Word < this[j].Word
+	}
+	return this[i].Freq < this[j].Freq
+}
+
+func (this PairList) Swap(i, j int) {
```
Comment (diwant) [inline exercise-001-corpus/corpus/corpus.go]: All exported functions like this Swap need a comment line above
